[CC-18] Set up `@cc/ui-chakra` and `@cc/ui-tailwind`

NorrisWu0 opened the pull request:
### New Feature
- [x] Set up `@cc/ui-chakra`, a react UI library with `@chakra-ui` and `@storybook`.
- [x] Set up `@cc/ui-tailwind`, a react UI library with `tailwindcss`.
- [x] Install `@cc/ui-chakra` onto `galaxy` project.
- [x] Install `@cc/ui-tailwind` onto `startrek` project.

### Chore
- [x] Bump version of `@types/react` in `@cc/ui-chakra` to fix issue with type mismatch on `React.ReactNode`.
- [x] Amend eslint rule `no-console` to warning level.'

### Notes

Additional comments are made outside this pull request, see the list below.

1. Questionable placeholder URL used in button onClick events. @kritomomo 
This change has been reflected in [fix: incorrect placeholder url](https://github.com/courtcanva/cc-multiverse/pull/3/commits/052b97e5e46b0fe28f7da14225bcbde6d0bc0bb3)

2. Lack of clarity around the usage of @cc/ui-chakra and @cc/ui-tailwind
Fixed in commit [docs: clarify purpose of ui libraries](https://github.com/courtcanva/cc-multiverse/pull/3/commits/43d2b22373c78316aeaa24050a7977f26c7bc0f4) @kritomomo 

3. Use of [interim type definition - VFC](https://basarat.gitbook.io/typescript/tsx/react#void-function-components) in components @kritomomo 
Updated as per comment in [refactor: remove storybook from @cc/ui-tailwind](https://github.com/courtcanva/cc-multiverse/pull/3/commits/ef2c802f9297e8f134a82fda2d48b983d608e3eb)

### Known issue
1. `storybook` currently does not work with `@cc/ui-tailwind`, however, the components themselves work as expected in `startrek` project.

tatoMa commented on apps/galaxy/src/pages/index.tsx:
```diff
@@ -0,0 +1,16 @@
+import { Stack, Button } from "@cc/ui-chakra";
+
+const NEVER_GONNA_GIVE_YOU_UP = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";
```
Serious?

NorrisWu0 replied:
Idea brought from @kritomomo 😂

tatoMa replied:
So how can I approve this pr?

NorrisWu0 replied:
Almost home, will fix this asap. 👌

NorrisWu0 commented:
Okay, since this PR is opened, there have been these changes made to these comments...

> Questionable placeholder URL used in button onClick events.
This change has been reflected in [fix: incorrect placeholder url](https://github.com/courtcanva/cc-multiverse/pull/3/commits/052b97e5e46b0fe28f7da14225bcbde6d0bc0bb3)

> Lack of clarity around the usage of `@cc/ui-chakra` and `@cc/ui-tailwind`
Fixed in commit [docs: clarify purpose of ui libraries](https://github.com/courtcanva/cc-multiverse/pull/3/commits/43d2b22373c78316aeaa24050a7977f26c7bc0f4)

> Use of [interim type definition - VFC](https://basarat.gitbook.io/typescript/tsx/react#void-function-components) in components
Updated as per comment in [refactor: remove storybook from @cc/ui-tailwind](https://github.com/courtcanva/cc-multiverse/pull/3/commits/ef2c802f9297e8f134a82fda2d48b983d608e3eb)

But yea it should be good to be reviewed again now. 👍 @tatoMa

chui-mel commented:
> Okay, since this PR is opened, there have been these changes made to these comments...
> 
> > Questionable placeholder URL used in button onClick events.
> 
> This change has been reflected in [fix: incorrect placeholder url](https://github.com/courtcanva/cc-multiverse/pull/3/commits/052b97e5e46b0fe28f7da14225bcbde6d0bc0bb3)
> 
> > Lack of clarity around the usage of `@cc/ui-chakra` and `@cc/ui-tailwind`
> 
> Fixed in commit [docs: clarify purpose of ui libraries](https://github.com/courtcanva/cc-multiverse/pull/3/commits/43d2b22373c78316aeaa24050a7977f26c7bc0f4)
> 
> > Use of [interim type definition - VFC](https://basarat.gitbook.io/typescript/tsx/react#void-function-components) in components
> 
> Updated as per comment in [refactor: remove storybook from @cc/ui-tailwind](https://github.com/courtcanva/cc-multiverse/pull/3/commits/ef2c802f9297e8f134a82fda2d48b983d608e3eb)
> 
> But yea it should be good to be reviewed again now. 👍 @tatoMa

can you move this part of information to PR description?

kritomomo commented on packages/ui-tailwind/src/display/Icon/Icon.tsx:
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+import NotFoundIcon from "@/assets/icons/404.svg";
```
It looks like you haven't used this icon inside the component

NorrisWu0 replied:
Okay... That's a mistake here, I think I will keep this here for now, I should be configuring webpack to load svg in the next pr (assuming I'm understanding this correctly)

kritomomo commented on packages/ui-tailwind/src/display/Icon/Icon.tsx:
```diff
@@ -0,0 +1,29 @@
+          d="M40 0C17.9032 0 0 17.9032 0 40C0 62.0968 17.9032 80 40 80C62.0968 80 80 62.0968 80 40C80 17.9032 62.0968 0 40 0ZM40 72.2581C22.2097 72.2581 7.74194 57.7903 7.74194 40C7.74194 22.2097 22.2097 7.74194 40 7.74194C57.7903 7.74194 72.2581 22.2097 72.2581 40C72.2581 57.7903 57.7903 72.2581 40 72.2581ZM27.0968 37.4194C29.9516 37.4194 32.2581 35.1129 32.2581 32.2581C32.2581 29.4032 29.9516 2 ...
+          fillOpacity="0.8"
+        />
+      </svg>
```
Can you replace this part with 404.svg, and in the next ticket, you may need to configure this https://github.com/courtcanva/cc-app/commit/32195a1c8d4dffd395f60ad670cc8dd5912ca4ad

NorrisWu0 replied:
Kool kool, will make sure this is configured in next pr.

kritomomo commented:
Regarding the classification of components in UI Library, at least at the code level, I think Button should not be classified in Form. I took a look at chakra's repo, which puts all components in the `components` folder. [chakra repo](https://github.com/chakra-ui/chakra-ui/tree/main/packages/components)
<img width="974" alt="image" src="https://user-images.githubusercontent.com/61776449/192425523-076eed30-a12f-4cad-8e67-fd182f079e6f.png">
For the material ui, they put the button into the base folder
<img width="884" alt="image" src="https://user-images.githubusercontent.com/61776449/192425938-01667fa8-5fcc-4ba2-bc99-2e6a78ca9f36.png">
Maybe this part need @tatoMa ‘s suggestion

tatoMa commented:
> Regarding the classification of components in UI Library, at least at the code level, I think Button should not be classified in Form. I took a look at chakra's repo, which puts all components in the `components` folder. [chakra repo](https://github.com/chakra-ui/chakra-ui/tree/main/packages/components) <img alt="image" width="974" src="https://user-images.githubusercontent.com/61776449/192425523-076eed30-a12f-4cad-8e67-fd182f079e6f.png"> For the material ui, they put the button into the base folder <img alt="image" width="884" src="https://user-images.githubusercontent.com/61776449/192425938-01667fa8-5fcc-4ba2-bc99-2e6a78ca9f36.png"> Maybe this part need @tatoMa ‘s suggestion

Totally agree with you. My opinion is button components can be inside of the base components. But it is not set in stone though.

tatoMa commented on apps/startrek/src/styles/globals.css:
```diff
@@ -0,0 +1,17 @@
+
+@layer base {
+  html {
+    font-family: Lexend, sans-serif;
```
Are we using this font or it is a placeholder?

NorrisWu0 replied:
This font was not referenced from our UI, it's just a placeholder font to check if the configuration is correct in tailwindcss

tatoMa commented on packages/ui-chakra/src/form/Button/Button.tsx:
```diff
@@ -0,0 +1,35 @@
+  };
+
+  return (
+    <CKButton onClick={onClick} color="#FFFFFF" {...buttonVariants[variant]}>
+      {children}
+    </CKButton>
```
Do you think we need useRef in the Button component or just leave it in the future if needed?

NorrisWu0 replied:
Hmm, I haven't had the experience to use ref for buttons yet, but yea, we can add them when we need to I think.
